prov/sm2/src/sm2_ep.c

```diff
@@ -273,6 +273,42 @@ static ssize_t sm2_do_inject(struct sm2_ep *ep, struct sm2_region *peer_smr,
 	return FI_SUCCESS;
 }
 
+static void sm2_format_cma(struct sm2_xfer_entry *xfer_entry,
+			   const struct iovec *iov, size_t count)
+{
+	struct sm2_cma_data *cma_data =
+		(struct sm2_cma_data *) xfer_entry->user_data;
+
+	/* All CMA messages must be delivery complete to stop completion from
+	 * being written instantly */
```

Comment:
We discussed this a little bit today.  My opinion on this matter is that setting op_flags to be FI_DELIVERY_COMPLETE is different than saying we can't generate completion until we get the xfer_entry back.  I think we can know this simply based on the protocol.

You aren't setting op_flags here, but this comment is misleading.

Reply:
If we don't set FI_DELIVERY_COMPLETE, then we will have to check for the protocol while writing completion entries. I end up having to do that anyway for the CUDA host-device protocol but Seth was concerned about additional if statements in the critical path. Personally, I'm fine either way.